### PR TITLE
Sbendus - added rozetka test

### PR DESCRIPTION
**Pull request by SofiiaPro**

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/HomePage.java [hunk 1]
```diff
@@ -1,17 +1,16 @@
 package com.softserveinc.ita.rozetka.page_objects;
 
+import lombok.Getter;
+
 import static com.codeborne.selenide.Selenide.$x;
 import static com.codeborne.selenide.Selenide.open;
 import static java.lang.String.format;
 
+@Getter
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/HomePage.java]: better use specifically on field

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 2]
```diff
@@ -53,13 +52,70 @@ public void testResultsOnProductsListPage() {
...
                 format("Middle product should be '%s'", expectedProductType));
-        assertTrue(productListPage.getProductNameForProduct(productCount).contains(expectedProductType),
+        assertTrue(productListPage.getProductName(productCount).contains(expectedProductType),
                 format("Last product should be '%s'", expectedProductType));
     }
+
+    @Test
+    public void testSelectProductByCatalog() {
+        var pageCategoryName = "Ноутбуки";
+        var chosenCategoryPageByCatalog = rozetkaHomePage
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 3]
```diff
@@ -53,13 +52,70 @@ public void testResultsOnProductsListPage() {
...
+                .getLeftSidebar()
+                .openCategory(pageCategoryName);
+        var categoryTitle = "Комп'ютери";
+        assertTrue(chosenCategoryPageByCatalog.getCategoryTitle().contains(categoryTitle),
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: split into var
newlines on 3 methods

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 4]
```diff
@@ -53,13 +52,70 @@ public void testResultsOnProductsListPage() {
...
+                .contains("Ноутбук Apple MacBook");
+        assertTrue(isProductTitleCorrect, "Incorrect product title");
+        var chosenCategoryPageByLink = chosenProductPage
+                .openCategoryPageByName(pageCategoryName.toLowerCase());
+        assertTrue(chosenCategoryPageByLink.getCategoryTitle().contains(categoryTitle),
+                "Incorrect page title");
+    }
+
+    @Test
+    public void testCompareProductPrices() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: rename, be specific

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 5]
```diff
@@ -53,13 +52,70 @@ public void testResultsOnProductsListPage() {
...
+        var catalogCategoryPage = rozetkaHomePage
+                .getLeftSidebar()
+                .openCategory("Знижки");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: Знижки to var

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/Catalog.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
+package com.softserveinc.ita.rozetka.page_objects;
+
+import com.codeborne.selenide.SelenideElement;
+
+import static com.codeborne.selenide.CollectionCondition.sizeGreaterThanOrEqual;
+import static com.codeborne.selenide.Selenide.*;
+import static java.lang.String.format;
+
+public class Catalog {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/Catalog.java]: *Page

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/Catalog.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+    public Catalog scrollToCategory(String categoryName) {
+        SelenideElement category = $x(format("//ul[@class='menu-categories'] //a[contains(@class, 'menu-categories__link') and contains(text(), '%s')]", categoryName));
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/Catalog.java]: var

Comment (toleksyn) [reply src/main/java/com/softserveinc/ita/rozetka/page_objects/Catalog.java]: rename

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/Header.java [hunk 8]
```diff
@@ -18,4 +19,9 @@ public HamburgerBar openHamburgerBar() {
     public boolean isSearchButtonDisplayed() {
         return $x("//button[contains(@class, 'search-form__submit')]").isDisplayed();
     }
+
+    public Catalog openCatalog() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/Header.java]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 9]
```diff
@@ -60,5 +60,82 @@ public void testResultsOnProductsListPage() {
         assertTrue(productListPage.getProductName(productCount).contains(expectedProductType),
                 format("Last product should be '%s'", expectedProductType));
     }
+
+    @Test
+    public void testSelectProductByLeftSidebar() {
+        var pageCategoryName = "Ноутбуки";
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 10]
```diff
@@ -60,5 +60,82 @@ public void testResultsOnProductsListPage() {
...
+        var productPage = salesCategoryPage.openProductByNumber(1);
+        var preDiscountPrice = productPage.getPreDiscountPrice();
+        var discountPrice = productPage.getDiscountPrice();
+        assertTrue(preDiscountPrice > discountPrice,
+                "Pre discount price should be bigger than current price with discount");
+    }
+
+    @Test
+    public void testShowMoreProducts() {
+        var chosenProductPage = rozetkaHomePage
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: ??

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 11]
```diff
@@ -60,5 +60,82 @@ public void testResultsOnProductsListPage() {
...
+        var extendedLastProductName = chosenProductPage.getProductName(extendedProductCount);
+        assertTrue(firstProductName.equals(extendedFirstProductName),
+                "First product name should be the same to the first product name after the extended page");
+        assertTrue(!lastProductName.equals(extendedLastProductName),
+                "Last product name should be different to the last product name after the extended page");
+    }
+
+    @Test
+    public void testSelectProductByCatalog() {
+        var catalogDropDownMenu = rozetkaHomePage
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: rename

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/CatalogMenu.java [hunk 12]
```diff
@@ -0,0 +1,32 @@
+package com.softserveinc.ita.rozetka.page_objects;
+
+import static com.codeborne.selenide.CollectionCondition.sizeGreaterThanOrEqual;
+import static com.codeborne.selenide.Selenide.*;
+import static java.lang.String.format;
+
+public class CatalogMenu {
+    public CategoryPage openCategory(String categoryName) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/CatalogMenu.java]: openCategoryPage()

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/CatalogMenu.java [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+        return new CategoryPage();
+    }
+
+    public CatalogMenu scrollToCategory(String categoryName) {
+        var category = $x(format("//ul[@class='menu-categories'] //a[contains(@class, 'menu-categories__link') and contains(text(), '%s')]", categoryName));
+        actions().moveToElement(category).perform();
+        return this;
+    }
+
+    public int getCategoryListSize() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/CatalogMenu.java]: rename

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/CatalogMenu.java [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+        $x(format("//li//a[@class='menu__hidden-title' and contains(text(), '%s')]", productCategoryName)).click();
+        return new ProductsListPage();
+    }
+
+    public List<String> getCategoryNames() {
+        return $$x("//ul[contains(@class, 'menu-categories_type_main')] //a[contains(@class, 'menu-categories__link')]")
+                .shouldHave(sizeGreaterThanOrEqual(1))
+                .texts();
+    }
+
```
Comment (VladimirPetrat) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/CatalogMenu.java]: Delete

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/CatalogMenu.java [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+        return new CategoryPage();
+    }
+
+    public CatalogMenu scrollToCategory(String categoryName) {
+        var category = $x(format("//ul[@class='menu-categories'] //a[contains(@class, 'menu-categories__link') and contains(text(), '%s')]", categoryName));
+        actions().moveToElement(category).perform();
+        return this;
+    }
+
+    public int getCategoriesSize() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/CatalogMenu.java]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 16]
```diff
@@ -136,5 +133,82 @@ public void testNextPreviousPagePagination() {
         assertEquals(productListPage.getCurrentPageNumber(), currentPageNumber,
                 "Previous page number is incorrect");
     }
+
+    @Test
+    public void testSelectProductByLeftSidebar() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: rename

Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 17]
```diff
@@ -136,5 +133,82 @@ public void testNextPreviousPagePagination() {
...
+        var categoryName = "Ноутбуки";
+        var laptopCategoryPage = rozetkaHomePage
+                .getCatalogMenu()
+                .openCategoryPage(categoryName);
+        var categoryTitle = "Комп'ютери";
+        var isLaptopCategoryPageCorrect = laptopCategoryPage
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: *Opened()

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 18]
```diff
@@ -136,5 +133,82 @@ public void testNextPreviousPagePagination() {
...
+        assertTrue(isLaptopCategoryPageCorrect, "Incorrect page title");
+        var laptopProductPage = laptopCategoryPage
+                .openProductsListPage(categoryName)
+                .openProductByName("Ноутбук Apple MacBook Air 13");
+        var isProductTitleCorrect = laptopProductPage
+                .getProductTitle()
+                .contains("Ноутбук Apple MacBook");
+        assertTrue(isProductTitleCorrect, "Incorrect product title");
+        var laptopCategoryPageByLink = laptopProductPage
+                .openCategoryPageByName(categoryName.toLowerCase());
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: no new line
rename

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 19]
```diff
@@ -136,5 +133,82 @@ public void testNextPreviousPagePagination() {
...
+        var productsCount = productPage.getProductsAmount();
+        var firstProductName = productPage.getProductName(1);
+        var lastProductName = productPage.getProductName(productsCount);
+        productPage.showMoreProducts();
+        var extendedProductCount = productPage.getProductsAmount();
+        var extendedFirstProductName = productPage.getProductName(1);
+        var extendedLastProductName = productPage.getProductName(extendedProductCount);
+        assertTrue(firstProductName.equals(extendedFirstProductName),
+                "First product name should be the same to the first product name after the extended page");
+        assertTrue(!lastProductName.equals(extendedLastProductName),
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: assertFalse

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 20]
```diff
@@ -47,35 +48,35 @@ public ProductPage openProductTab(ProductPageTab productPageTab) {
...
+        return $x(format("//div[contains(@class, 'characteristics-full__item') and .//span/text()='%s']//li/*",
                 characteristicType)).text();
     }
 
-    public HomePage openHomePageByLogo() {
-        $x("//*[@class='header__logo']").click();
-        return new HomePage();
+    @Step("Product page: open cheaper-together menu")
+    public CheaperTogetherMenu openCheaperTogetherMenu() {
+        return new CheaperTogetherMenu();
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java:80]: there's no open action here, rename

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+                .getCategoryTitle()
+                .contains(categoryTitle);
+        assertTrue(isLaptopCategoryPageOpened, "Incorrect page title");
+    }
+
+    @Test
+    public void testCompareDiscountWithPreDiscountPrices() {
+        var categoryName = "Ціни";
+        var salesCategoryPage = catalogMenu
+                .openCategory(categoryName);
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java]: no new line

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/components/CheaperTogetherMenu.java [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+
+import static com.codeborne.selenide.Selenide.$x;
+import static java.lang.Integer.parseInt;
+import static java.lang.String.format;
+
+public class CheaperTogetherMenu extends ProductPage {
+
+    public String getProductNameByNumber(int number) {
+        return $x(format("(//li[contains(@class, 'kits-list__item_type_proposed')]//a[contains(@class, 'kits-tile__title')])[%d]", number))
+                .text();
```
Comment (VladimirPetrat) [inline src/main/java/com/softserveinc/ita/rozetka/components/CheaperTogetherMenu.java]: No need to new line

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java [hunk 23]
```diff
@@ -78,7 +78,6 @@ public ProductsListPage showMoreProducts() {
         return this;
     }
 
-    @Step("Products list page: get age confirmation popup")
     public AgeConfirmationPopup getAgeConfirmationPopup() {
```
Comment (VladimirPetrat) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductsListPage.java]: Make fetch

File: src/test/java/com/softserveinc/ita/rozetka_test/ProductNavigationTest.java [hunk 24]
```diff
@@ -51,9 +51,9 @@ public void testSortByPrice() {
                 .searchFor("сокира")
                 .setSortingType(SortingOption.CHEAP);
         var lastProductNumber = productListPage.getProductsAmount();
-        var firstProductPrice = productListPage.getPriceFromProduct(1);
-        var lastProductPrice = productListPage.getPriceFromProduct(lastProductNumber);
-        var middleProductPrice = productListPage.getPriceFromProduct(lastProductNumber / 2);
+        var firstProductPrice = productListPage.getProductPrice(1);
+        var lastProductPrice = productListPage.getProductPrice(lastProductNumber);
+        var middleProductPrice = productListPage.getProductPrice(lastProductNumber / 2);
         assertTrue(middleProductPrice < lastProductPrice,
```
Comment (VladimirPetrat) [inline src/test/java/com/softserveinc/ita/rozetka_test/ProductNavigationTest.java:57]: Fetch

File: src/test/java/com/softserveinc/ita/rozetka_test/ProductNavigationTest.java [hunk 25]
```diff
@@ -101,5 +101,18 @@ public void testNextPreviousPagePagination() {
         assertEquals(productListPage.getCurrentPageNumber(), currentPageNumber,
                 "Previous page number is incorrect");
     }
+
+    @Test
+    public void testSelectProductByCatalog() {
+        var catalogMenu = header.openCatalogMenu();
```
Comment (VladimirPetrat) [inline src/test/java/com/softserveinc/ita/rozetka_test/ProductNavigationTest.java]: Test situated in wrong class

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java [hunk 26]
```diff
@@ -49,4 +48,91 @@ public void testResultsOnProductsListPage() {
...
+                .contains(categoryTitle);
+        assertTrue(isLaptopCategoryPageOpened, "Incorrect page title");
+        var laptopProductPage = laptopCategoryPage
+                .openProductsListPage(categoryName)
+                .openProductByName("Ноутбук Apple MacBook Air 13");
+        var isProductTitleCorrect = laptopProductPage
+                .getProductTitle()
+                .contains("Ноутбук Apple MacBook");
+        assertTrue(isProductTitleCorrect, "Incorrect product title");
+        var laptopCategoryPageByLink = laptopProductPage.openCategoryPageByLink(categoryName.toLowerCase());
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java:68]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java [hunk 27]
```diff
@@ -49,4 +48,91 @@ public void testResultsOnProductsListPage() {
...
+
+    @Test
+    public void testSelectProductByCatalogMenu() {
+        var categoryName = "Ноутбуки";
+        var laptopCategoryPage = catalogMenu.openCategory(categoryName);
+        var categoryTitle = "Комп'ютери";
+        var isLaptopCategoryPageOpened = laptopCategoryPage
+                .getCategoryTitle()
+                .contains(categoryTitle);
+        assertTrue(isLaptopCategoryPageOpened, "Incorrect page title");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java:60]: fix

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java [hunk 28]
```diff
@@ -49,4 +48,91 @@ public void testResultsOnProductsListPage() {
...
+    @Test
+    public void testSelectProductByCatalog() {
+        var makeUpCategory = catalogMenu.scrollToCategory("Краса");
+        var subcategoryAmount = makeUpCategory.getCategoriesAmount();
+        assertTrue(subcategoryAmount > 0, "Category should contains at least one subcategory");
+        var productsListPage = makeUpCategory.openProductsListPage("Догляд за обличчям");
+        var isProductListPageOpened = productsListPage
+                .getPageTitle()
+                .contains("Косметика для догляду за обличчям");
+        assertTrue(isProductListPageOpened, "Incorrect page title");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java:136]: fix

Comment (toleksyn) [reply src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java:136]: this is not enough

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java [hunk 29]
```diff
@@ -49,4 +48,91 @@ public void testResultsOnProductsListPage() {
...
+        var discountPrice = cheaperTogetherMenu.getDiscountPrice(1);
+        var currentPrice = productPage.getDiscountPrice();
+        assertTrue(discountPrice < currentPrice,
+                "The discount price should be smaller than the current product price");
+    }
+
+    @Test
+    public void testSelectProductByCatalog() {
+        var makeUpCategory = catalogMenu.scrollToCategory("Краса");
+        var subcategoryAmount = makeUpCategory.getCategoriesAmount();
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java:130]: fix

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java (hunk 26 above)
Comment (toleksyn) [reply src/test/java/com/softserveinc/ita/rozetka_test/CategoryNavigationTests.java:68]: not sure

**Pull request closed without merging**